cmd/object-api-utils.go

```diff
@@ -64,12 +69,13 @@ func IsValidBucketName(bucket string) bool {
 	if len(bucket) < 3 || len(bucket) > 63 {
 		return false
 	}
-	if bucket[0] == '.' || bucket[len(bucket)-1] == '.' {
-		return false
+	labels := strings.Split(bucket, ".")
```

Comment:
Avoid saving into a variable since its not useful after this loop 

```
for _, label := range strings.Split(bucket, ".") {
}
```

Also lets keep `"."` as DNS delimiter constant to avoid mistakes.